src/python/pants/backend/jvm/subsystems/scala_platform.py

```diff
@@ -132,17 +138,37 @@ def register_custom_tool(key):
     register_custom_tool('scala-repl')
     register_custom_tool('scalastyle')
 
-  def _tool_classpath(self, tool, products):
+  def _tool_classpath(self, tool, products, scheduler):
     """Return the proper classpath based on products and scala version."""
-    return self.tool_classpath_from_products(products,
-                                             self._key_for_tool_version(tool, self.version),
-                                             scope=self.options_scope)
+    classpath = self.tool_classpath_from_products(products,
+                                                  self._key_for_tool_version(tool, self.version),
+                                                  scope=self.options_scope)
+    classpath = tuple(fast_relpath(c, get_buildroot()) for c in classpath)
+
+    return self._memoized_scalac_classpath(classpath, scheduler)
+
+  @memoized_method
+  def _memoized_scalac_classpath(self, scala_path, scheduler):
+    if scala_path:
+      snapshots = scheduler.capture_snapshots(tuple(PathGlobsAndRoot(PathGlobs([path]), get_buildroot()) for path in scala_path))
+      return [ClasspathEntry(path, snapshot) for path, snapshot in list(zip(scala_path, snapshots))]
+    else:
+      return [ClasspathEntry(p, None) for p in scala_path]
```

Comment:
In this branch of the if/else the `scala_path` will definitely be empty or `None`, so this case doesn't make sense.